aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -76,21 +99,51 @@
 
     private GetPolicyResponse getExistingPolicy(AmazonWebServicesClientProxy proxy, String profilingGroupName) {
         return proxy.injectCredentialsAndInvokeV2(
-            GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build(),
-            profilerClient::getPolicy
+                GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build(),
+                profilerClient::getPolicy
         );
     }
 
+    private void updateNotificationChannels(String pgName, AmazonWebServicesClientProxy proxy, List<Channel> currentChannels) {
+        List<Channel> existingChannels = getExistingNotificationConfiguration(proxy, pgName).notificationConfiguration().channels();
+        updateNotificationChannels(existingChannels, currentChannels, pgName, proxy);
```

Comment:
Having both `currentChannels` and `existingChannels` is a bit confusing. Perhaps `requestedConfiguration` and `currentChannels`?